src/Cardano/NetworkLayer.hs

```diff
@@ -43,7 +43,7 @@ data NetworkLayer m e0 e1 = NetworkLayer
 -- an action for each elements retrieved.
 tick
     :: forall st m b. (MonadIO m)
-    => (st -> m ([b], st))
+    => (st -> m (b, st))
```

Comment:
Does this assume that one tick will handle only one block? What when the tick frequency is lower than the frequency of the change of the tip of head?

Reply:
Nope, it means that the tick function doesn't assume anything about how the chain producer produces blocks. So, it passes around whatever it receives, if it's a list of 10K blocks, then it's a list of 10k blocks. If it's a single one, it will only pass a single one. 

So, `tick` is now much more general and just execute an action every `x` seconds.

Reply:
> Still, do not understand why we forbid using more than one block per tick ... All "gaps" are going to be tackled elsewhere?

We do not. We actually leave the type free / more flexible. Before, we forced that we had a list of `b` and that the action would take ONLY one `b`. In practice, that's rather unpractical because, the action may work better by handling batches of `b` (and avoid for instance, to do a DB lookup / write for each of them, but only modify the model in memory for a batch, and then, flush the result to DB!). So, now, the action just expects whatever was yielded by the chain producer which is in practice, a list of blocks (possibly a singleton when we are in sync with the node).

Reply:
thanks, now I see it